kafka/producer/producer_batch.go

```diff
@@ -116,7 +120,16 @@ func (b *Batch) FlushMessages() {
 			return
 		}
 		b.metric.BatchProduceLatency = time.Since(startedTime).Milliseconds()
-
+		if b.callback != nil {
+			for _, msg := range b.messages {
+				go b.callback(message.KafkaMessage{
```

Comment:
I think it should be sync what do you think, we may miss some callback

Reply:
It depends on the situation I prefer an async callback function because I don't want to block the event publish because of the callback function. But you're right in your concern, let's convert it to sync by default, and maybe in another development, we can leave it up to the user with a flag to decide whether they want to call it asynchronously or synchronously.